Eevee.Sleep.Bot/Handlers/EventHandlers/GuildMemberUpdatedEventHandler.cs

```diff
@@ -135,12 +135,19 @@ SocketGuildUser updated
             .Where(x => rolesAdded.Contains(x.RoleId))
             .ToHashSet();
 
+        if (rolesAdded.Length > 0)
+        {
+            await UserRoleContoller.AddRoles(updated.Id.ToString(), rolesAdded);
```

Comment:
Remember to check if the role is tracked before adding the role, maybe have a method similar to `HandleRolesAdded()` here and do all the works? Checking impl shouldn't be in the controller impl.

Also if needed rename `HandleRolesAdded()` to whatever suitable to avoid confusion.

Reply:
In this case, need DB access to check if a system role is already tracked, do I need to check here?
It didn't allow duplicates when the UPDATE, so I didn't think It needed to check here.

Reply:
If you mean it needs a controller method to check if the role(s) is/are being tracked, then yes
For avoiding duplicate, using `$addToSet` (the changes you made regarding `$addToSet` `$pull` basically) should be fine.
Checking if a role is being tracked or not will be in a separate collection.

Reply:
I made sure to check it out first.
[042bf5a](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/042bf5ae19def3a76b4fa219af2021bdffdd459e)

```
    private static async Task HandleRolesAdded(ulong userId, ulong[] addedRoles) {
        var roles = DiscordRoleContoller.FindRoleIdsByUserId(userId);
        var untrackedRole = roles.Except(addedRoles).ToArray();

        if (untrackedRole.Length > 0) {
            await DiscordRoleContoller.AddRoles(userId, untrackedRole);
        }
    }
```